include/metaprogram.h

```diff
@@ -2,6 +2,15 @@
 #ifndef METAPROGRAM_H
 #define METAPROGRAM_H
 
+/* Check if VA_OPT_ is supported by the compiler. GCC's version should be at least 9.5*/
+#define PP_THIRD_ARG(a,b,c,...) c
+#define VA_OPT_SUPPORTED_I(...) PP_THIRD_ARG(__VA_OPT__(,),true,false,)
```

Comment:
I think this might need to be `TRUE` and `FALSE`? :thinking: 
```suggestion
#define VA_OPT_SUPPORTED_I(...) PP_THIRD_ARG(__VA_OPT__(,),TRUE,FALSE,)
```